kroxylicious-kms/src/main/java/io/kroxylicious/kms/service/DestroyableRawSecretKey.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.service;
+
+import java.security.MessageDigest;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Objects;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import javax.crypto.SecretKey;
+import javax.security.auth.DestroyFailedException;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A SecretKey that has RAW encoding and can be {@linkplain #destroy() destroyed}
+ * (unlike {@link javax.crypto.spec.SecretKeySpec}).
+ */
+@NotThreadSafe
+public final class DestroyableRawSecretKey implements SecretKey {
```

Comment:
`getEncoded` is defined on `java.security.Key`, so _all_ keys are encodable some how. So I don't think there's really anything to be gained by making encodable part of the class name. We do currently have 'raw' in there, which is just mimicing what `SecretKeySpec` does. I could live with `DestroyableSecretKeySpec`, I guess.

Reply:
Not all SecretKeys have an encoded form.  The javadoc for java.security.Key#getEncoded:

>Returns the key in its primary encoding format, or null if this key does not support encoding.
> Returns:
> the encoded key, or null if the key does not support encoding.

Currently `io.kroxylicious.kms.service.DestroyableRawSecretKey#toDestroyableKey` accepts any SecretKey that has an encoded form.   The Objects.requireNotNull will reject any `SecretKey` where `getEncoded() == null`.  That was what prompted the idea for `Encodable` to be part of the name rather than the word `Raw`.  I thought that was more precise.

However, perhaps the real issue is the fact that `DestroyableRawSecretKey#toDestroyableKey()` will accept keys with encoding format other than `RAW`.  Either DestroyableRawSecretKey#getFormat should return the actual format of the wrapped key, or `DestroyableRawSecretKey#toDestroyableKey()` should insist that the source SecretKey has type `RAW`.  In the latter case, the class name is accurate.